plenum/server/propagator.py

```diff
@@ -81,6 +82,15 @@ def forwarded(self, req: Request) -> bool:
         """
         return self[req.key].forwarded
 
+    def ordered_by_replica(self, request_key):
```

Comment:
I think this also needs to be called in the following cases:
1) `_clear_request_for_txn`
2) `clear_requests_and_fix_last_ordered`